### DISCUSSION TITLE
Using SelectFilter to query for field with NULL values?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#5611] [Help]
Hello,

I am trying to add an option to a SelectFilter for records with NULL values.  I figured out how to add my custom row to the options but cannot get it to query correctly.

```
SelectFilter::make('noderole')
                    ->relationship('noderole', 'name')
                    ->label('Node Role')
                    ->options(function () {
                        $results = NodeRole::all()->pluck('name', 'id');
                        $results->prepend('None', 'null');
                        return $results;
                    }),
```

This works to update the browser parameters `&tableFilters[noderole][value]=null` however it does not return the correct results.

**Comment by ArtMin96** [comment]
It doesn't work because you send the string "null", which is not the unit type of php

**Reply by yaplej** [reply]
It looks like the SelectFilter would have to be modified to detect a special string and convert it to the proper null to use in the select statement.  I don't see any other way to do this.
